src/features/ProfilePage/components/PublicInfo.tsx

```diff
@@ -2,34 +2,101 @@ import styled from 'styled-components';
 import { useTranslation } from 'react-i18next';
 
-import { selectUserInfo } from '@/features/Authentication/userSlice';
-import { useAppSelector } from '@/store';
-
-import { DEFAULT_ICON_SIZE, palette } from '@/components/constants';
+import {
+  selectMentor,
+  selectUser,
+  setMentor,
+  setUser,
+} from '@/features/Authentication/userSlice';
+import { useAppDispatch, useAppSelector } from '@/store';
+import { useDebounce } from '@/hooks/useDebounce';
+
+import {
+  DEFAULT_ICON_SIZE,
+  palette,
+  SAVE_DELAY_MS,
+} from '@/components/constants';
 import LabeledInput from '@/components/LabeledInput';
 import Slider from '@/components/Slider';
 import Text from '@/components/Text';
 import TextInput from '@/components/TextInput';
+import {
+  useUpdateMentorMutation,
+  useUpdateUserMutation,
+} from '../profilePageApi';
+
+import type { Mentor, User } from '@/features/Authentication/authenticationApi';
 
 const PublicInfo = () => {
   const { t } = useTranslation('profile');
-  const userInfo = useAppSelector(selectUserInfo);
-
-  const [displayName, setDisplayName] = useState('');
-  const [birthYear, setBirthYear] = useState('');
-  const [area, setArea] = useState('');
-  const [status, setStatus] = useState('');
-  const [isActive, setIsActive] = useState(true);
-  const [story, setStory] = useState('');
+  const dispatch = useAppDispatch();
+  const user = useAppSelector(selectUser);
+  const mentor = useAppSelector(selectMentor);
+  const [updateUser, { isError: isUserError, isSuccess: isUserSuccess }] =
+    useUpdateUserMutation();
+  const [updateMentor, { isError: isMentorError, isSuccess: isMentorSuccess }] =
+    useUpdateMentorMutation();
+
+  const [birthYear, setBirthYear] = useState(String(mentor.birth_year));
+  const [displayName, setDisplayName] = useState(mentor.display_name);
+  const [isAbsent, setIsAbsent] = useState(mentor.is_vacationing);
+  const [region, setRegion] = useState(mentor.region);
+  const [status, setStatus] = useState(mentor.status_message);
+  const [story, setStory] = useState(mentor.story);
   const [topicSearchValue, setTopicSearchValue] = useState('');
 
-  const toggleIsActive = () => setIsActive(isActive => !isActive);
+  const toggleIsAbsent = () => setIsAbsent(!isAbsent);
+
+  const userToSave: User = {
+    ...user,
+    display_name: displayName,
+  };
+
+  const mentorToSave: Mentor = {
+    ...mentor,
+    birth_year: Number(birthYear),
+    display_name: displayName,
+    is_vacationing: isAbsent,
+    region,
+    status_message: status,
+    story,
+  };
+
+  useEffect(() => {
+    if (isUserError || isMentorError) {
+      // TODO: Show error notification
+    }
+  }, [isUserError, isMentorError]);
+
+  useEffect(() => {
+    if (isUserSuccess) dispatch(setUser(userToSave));
+  }, [isUserSuccess]);
+
+  useEffect(() => {
+    if (isMentorSuccess) {
+      dispatch(setMentor(mentorToSave));
+      // TODO: Show success notification
+    }
+  }, [isMentorSuccess]);
```

Comment:
No.. :D 
no effect trains please!
This file looks very confusing and I have almost no idea what its supposed to do, even when I know what its supposed to do.

First of all, we should leverage the rtk-mutations api.
1. We do a mutation
2. We invalidate the query
3. We automatically fetch the updated data

This is done with `tagTypes`, `providesTags` and `invalidatesTags`
https://redux-toolkit.js.org/rtk-query/usage/automated-refetching

```typescript
// some api providesTags, like getMentors
export const mentorsApi = createApi({
  baseQuery: refreshingBaseQuery,
  reducerPath: 'mentors',
  tagTypes: ['mentors'],

  endpoints: builder => ({
    getMentors: builder.query<Mentors, void>({
      query: () => 'mentors',
      providesTags: ['mentors'],
      transformResponse: (response: unknown) =>
....

// in the user-api we can invalidateTagTypes 'tagTypes'
export const profilePageApi = createApi({
  baseQuery: refreshingBaseQuery,
  reducerPath: 'profile',
  tagTypes: ['user, account', 'mentors'], // note that in different API-clients we need to have the same-tagTypes again
.......
    updateMentor: builder.mutation<unknown, Mentor>({
      query: mentor => ({
        url: `/mentors/${mentor.id}`,
        method: 'put',
        body: mentor,
      }),
      invalidatesTags: ['mentors'],
    }),


  // we only launch the async-operations, and we can chain them yes, no effect
  const saveUserAndMentor = useDebounce(() => {
    updateUser(userToSave).then(() => {
      return updateMentor(mentorToSave);
    });
  }, SAVE_DELAY_MS);
```

So when we getMentors, its ok. Then we update the profile, and we can just fetch the mentors again automatically. No need for funny effects. Everything will be mapped back to the UI

Reply:
It's done and works like a charm!